arcgis-ios-sdk-samples/Scenes/Animate image overlay/AnimateImagesWithImageOverlayViewController.swift

```diff
@@ -0,0 +1,235 @@
+    // MARK: Storyboard views
+    
+    /// The button to play image overlay animation.
+    @IBOutlet var playButtonItem: UIBarButtonItem!
+    /// The button to pause image overlay animation.
+    @IBOutlet var pauseButtonItem: UIBarButtonItem!
+    /// The button to choose a frame rate for the animation
+    @IBOutlet weak var speedButtonItem: UIBarButtonItem!
+    /// The toolbar in the view controller.
+    @IBOutlet weak var toolbar: UIToolbar!
+    /// The label to display opacity level.
+    @IBOutlet weak var opacityLabel: UILabel!
+    
+    /// The slider to change opacity level, from transparent 0% to opaque 100%.
+    @IBOutlet weak var opacitySlider: UISlider! {
+        didSet {
+            imageOverlay.opacity = opacitySlider.value
+            opacityLabel.text = getOpacityPercentageString(percentage: opacitySlider.value)
+        }
+    }
+    
+    /// The scene view managed by the view controller.
+    @IBOutlet weak var sceneView: AGSSceneView! {
+        didSet {
+            sceneView.scene = makeScene()
+            let point = AGSPoint(x: -116.621, y: 24.7773, z: 856977.0, spatialReference: .wgs84())
+            let camera = AGSCamera(location: point, heading: 353.994, pitch: 48.5495, roll: 0)
+            sceneView.setViewpointCamera(camera)
+            sceneView.imageOverlays.add(imageOverlay)
+        }
+    }
+    
+    // MARK: Instance properties
+    
+    /// The frame rate in the unit of frame per second.
+    var frameRate = 60
+    /// A timer to synchronize image frame animation to the refresh rate of the display.
+    var displaylink: CADisplayLink?
+    /// The image overlay to show image frames.
+    let imageOverlay = AGSImageOverlay()
+    /// An iterator to hold and loop through the overlay images.
+    private var imagesIterator: CircularIterator<UIImage>!
+    
+    /// A formatter to format percentage strings.
+    let percentageFormatter: NumberFormatter = {
+        let formatter = NumberFormatter()
+        formatter.numberStyle = .percent
+        formatter.multiplier = 100
+        return formatter
+    }()
+    
+    /// An envelope of the pacific southwest sector for displaying the image frame.
+    let pacificSouthwestEnvelope = AGSEnvelope(
+        center: AGSPoint(
+            x: -120.0724273439448,
+            y: 35.131016955536694,
+            spatialReference: .wgs84()
+        ),
+        width: 15.09589635986124,
+        height: -14.3770441522488
+    )
+
+    /// A boolean which indicates whether the animation is playing or paused.
+    var isAnimating = false {
+        didSet {
+            if isAnimating {
+                let index = toolbar.items!.firstIndex(of: playButtonItem)!
+                toolbar.items![index] = pauseButtonItem
+            } else {
+                let index = toolbar.items!.firstIndex(of: pauseButtonItem)!
+                toolbar.items![index] = playButtonItem
+            }
+        }
```

Comment:
This is a bit cumbersome, but I'm trying to avoid modifying the bar button at a certain index.